lib/kubernetes-deploy/kubernetes_resource/pod.rb

```diff
@@ -80,28 +86,25 @@ def failure_message
       "#{phase_problem}#{container_problems}".presence
     end
 
-    # Returns a hash in the following format:
-    # {
-    #   "app" => ["array of log lines", "received from app container"],
-    #   "nginx" => ["array of log lines", "received from nginx container"]
-    # }
-    def fetch_logs(kubectl)
-      return {} unless exists? && @containers.present?
-      @containers.each_with_object({}) do |container, container_logs|
-        cmd = [
-          "logs",
-          @name,
-          "--container=#{container.name}",
-          "--since-time=#{@deploy_started_at.to_datetime.rfc3339}",
-        ]
-        cmd << "--tail=#{LOG_LINE_COUNT}" unless unmanaged?
-        out, _err, _st = kubectl.run(*cmd, log_failure: false)
-        container_logs[container.name] = out.split("\n")
-      end
+    def fetch_debug_logs(kubectl)
+      logs.sync(kubectl)
+      logs
+    end
+
+    def supports_debug_logs?
+      exists? && !@stream_logs # don't print them a second time
```

Comment:
Nitpick: This seems like a bit of a hack to me.. Just because the logs have already been printed doesn't mean that this thing doesn't support logging anymore. Seems like this check should either be in a different place or this method should have a different name (`print_debug_logs?` maybe)?

Reply:
`print_debug_logs?` works for me